step-controller/step-controller-server/src/main/java/step/resources/ResourceImporter.java

```diff
@@ -45,54 +50,51 @@ public void accept(Object t, ImportContext importContext) {
+			AtomicReference<ResourceRevision> currentRevision = new AtomicReference<>(null);
+			resourceRevisionAbstractAccessor.getResourceRevisionsByResourceId(origResourceId).forEachRemaining(resourceRevision -> {
+				//Make sure we create the resource's current revision at the end
+				if (!tmpCollectionCurrentRevisionId.equals(resourceRevision.getId().toHexString())) {
+					importRevision(importContext, resourceRevision, resource, origResourceId, importConfiguration);
+				} else {
+					currentRevision.set(resourceRevision);
+				}
+			});
+			//Create current revision
+			if (currentRevision.get() != null) {
+				importRevision(importContext, currentRevision.get(), resource, origResourceId, importConfiguration);
+			} else {
+				throw new RuntimeException("Error while updating resource content for resource " + origResourceId + ": the current revision " + tmpCollectionCurrentRevisionId + " was not found.");
 			}
 		}
 	}
 
-	public void saveCopyOfResourceToLocalResourceManager(LocalResourceManagerImpl localResourceMgr, Resource resource,
-			String origResourceId) {
-		try {
-			Resource resource2 = new Resource();
-
-			resource2.setCreationDate(resource.getCreationDate());
-			resource2.setCreationUser(resource.getCreationUser());
-			resource2.setLastModificationDate(resource.getLastModificationDate());
-			resource2.setLastModificationUser(resource.getLastModificationUser());
-
-			resource2.setResourceName(resource.getResourceName());
-			resource2.addAttribute(AbstractOrganizableObject.NAME, resource.getResourceName());
-			resource2.setId(new ObjectId(origResourceId));
-			resource2.setCurrentRevisionId(resource.getCurrentRevisionId());
-			resource2.setResourceType(resource.getResourceType());
-			localResourceMgr.saveResource(resource2);
-			
-			ObjectId currentRevisionId = resource.getCurrentRevisionId();
-			ResourceRevision revision = new ResourceRevision();
-			revision.setId(currentRevisionId);
-			revision.setResourceId(resource2.getId().toString());
-			revision.setResourceFileName(resource2.getResourceName());
-			localResourceMgr.saveResourceRevision(revision);
-		} catch (IOException e) {
-			throw new RuntimeException("Unexpected error while creating temporary resource for resource "+origResourceId, e);
+	private void importRevision(ImportContext importContext, ResourceRevision resourceRevision, Resource resource, String origResourceId, ImportConfiguration importConfiguration) {
+		String origRevisionId = importConfiguration.isOverwrite() ? resourceRevision.getId().toHexString() :
+				importContext.getNewToOldReferences().get(resourceRevision.getId().toHexString());
+		Path path = Paths.get(importContext.getWorkFolder().getAbsolutePath(), resource.getResourceType(), origResourceId, origRevisionId, resourceRevision.getResourceFileName());
```

Comment:
Ideally, the construction of the Path should be delegated to the ResourceManager. If we would change the structure of how files are stored, this woulnd't work otherwise